src/main.rs

```diff
@@ -504,29 +497,29 @@ impl AsyncComponent for App {
             gtk::glib::Propagation::Stop
         });
 
-        #[cfg(target_os = "linux")]
-        let tray_img = Icon::from_png_bytes(TRAY_ICON).expect("Tray icon must exist");
+        #[cfg(any(target_os = "macos", target_os = "linux"))]
+        let tray_img = Icon::from_png_bytes(TRAY_ICON).expect("Tray icon exists and is a valid PNG; qe");
         #[cfg(target_os = "windows")]
-        let tray_img = Icon::from_resource(1, None).expect("Tray icon must exist");
+        let tray_img = Icon::from_resource(1, None).expect("Tray icon exists and is a valid PNG; qe");
 
         let tray = TrayIconBuilder::new()
             .with_icon(tray_img)
             .with_tooltip("Space Acres")
-            .with_menu(build_menu())
+            .with_menu(Menu::new([
+                MenuItem::button("Open", TrayMenuSignal::Open),
+                MenuItem::button("Quit", TrayMenuSignal::Quit),
+            ]))
```

Comment:
Please read suggestions very carefully. What I suggested for previous code revision was to use `AppInput` instead of `TrayMenuSignal`, so you both simplify it here and don't need to match/map signals below.

Reply:
but the library request signal to be clone, and to make AppInput Clone change many relative types to be clone, and i see this type also involve not clone type in other lib.

```rs
impl<T: Clone + Send + 'static> TrayIconBuilder<T> {

    /// Attempts to create the tray icon. See the the *Platform notes* section of the Readme for more information.
    pub fn build<F>(self, callback: F) -> TrayResult<TrayIcon<T>>
        where F: FnMut(TrayEvent<T>) + Send + 'static
    {
        Ok(TrayIcon(NativeTrayIcon::new(self, callback)?))
    }
}
```